### ISSUE TITLE
Investigate 1ML

### ISSUE THREAD

**Issue by lpil** [lpil/gleam#22]
https://github.com/alpaca-lang/alpaca/issues/251#issuecomment-411104084

**Comment by rizo** [comment]
+💯

**Comment by lpil** [comment]
@rizo got any reading materials for me?

**Comment by rizo** [comment]
Not really, the only resources you'll find on 1ML is the original paper and this [prototype implementation](https://github.com/rossberg/1ml).
